core/toolbox.js

```diff
@@ -588,6 +589,83 @@ Blockly.Toolbox.TreeNode.prototype.onDoubleClick_ = function(e) {
 };
 
 /**
+ * Modify original onKeyDown handler to swap
+ * LEFT/RIGHT keys with UP/DOWN in horizontalLayout.
+ * @param {!goog.events.BrowserEvent} e The browser event.
+ * @return {boolean} The handled value.
+ * @override
```

Comment:
Wouldn't it be much simpler (in the case of a horizontal layout) to just modify the event's keycode before passing up to the existing goog.ui.tree.BaseNode.prototype.onKeyDown method?  Then we don't need to copy any of this code.